docs/architecture/widget_framework.md

```diff
@@ -0,0 +1,90 @@
+
+## Data flow for an audio readout request
+
+Audio equivalence support is built into the framework.
+
+Its data flow is about the same as the visual UI, but instead of being serialized to JSON and sent to the browser to be rendered by React, it is instead rendered to an SSML string on the server and piped through AWS Polly to produce an MP3 containing synthesized speech.
+
+1. Client requests `/v2/audio/{id}/readout.mp3`. Subsequent steps take place on the server.
+1. Steps **2** through **8** of the above data flow run as usual.
+1. The data map is rendered to SSML using each widget's audio view.
+1. The SSML is sent to Polly for synthesis, and we receive MP3 data in the response.
+1. The MP3 is sent back to the client for playback.
+
+## Anatomy of a template
+
+Each screen type has a template that defines the "slots" that widgets can fit into, as well as layout variations for special cases like high-priority alerts that take over all space below the screen header.
+
+A template is a recursive data structure.
+
+The base case is a "slot ID"—a simple named space that a widget can drop into.
+
+The recursive case is a tuple representing a region containing one or more layout variations: `{region_id, %{layout_id => list(template)}}`. Let's break that down:
+
+- `region_id`: the region's name. For example: `:body`, `:flex_zone`, `:screen`.
+- `layout_id`: a name for one possible set of slots that can fill this region. For example: `:body_normal`, `:body_takeover`; `:one_large`, `:two_medium`.
+- `list(template)`: the set of slots, or even further nested regions, that fill this region under the given layout variation.
+
+Layout variations provide a mechanism for high-priority widgets, like urgent alerts, to take over large portions of a screen. They also allow paged regions to hold widgets of different sizes per page.
+
+## Feature: Paging
+
+Some screen types have paged regions, where the widgets that appear in the region can rotate over the course of several data refreshes.
+
+Paging logic lives on the backend. The current page is selected based on the screen type's data refresh rate (e.g. 20 sec for pre-fare duo) and the current clock time.
+
+During the serialization step, all pages except the current one are discarded. This means the React client is only aware of the content of the current page at any given moment.
+
+### Why backend paging?
+
+Because audio readouts are not space constrained in the same way as the visual UI, and because they happen less frequently, we want to read out all info from a screen, including currently hidden pages. Since audio synthesis happens on the backend, this means the backend needs to be aware of, and in control of, paging.
+
+## Feature: Audio-only widgets
+
+Because the semantics of speech don't allow for the same "scanability" as a visual UI, we occasiona5lly need to provide a summary of content before it's read out, so that the rider can decide whether it's relevant to them.
```

Comment:
Just a small typo
```suggestion
Because the semantics of speech don't allow for the same "scanability" as a visual UI, we occasionally need to provide a summary of content before it's read out, so that the rider can decide whether it's relevant to them.
```